### ISSUE TITLE
Executing A Playbook That Calls The Vaulted Variable File Shows The Value Of Encrypted String In Job Output

### ISSUE THREAD

**Issue by Ompragash** [ansible/ansible-runner#354]
## Summary:

Executing a playbook that calls the vaulted variable file shows the value of encrypted string in the Job output.

## Enviroment:

Ansible Tower: v3.5.2
Ansible Engine: v2.8.4

## Steps to Reproduce

1. Encrypt the value `bar` and store it in a variable file.
**Note**: Vault password is **ansible**

~~~
cat string.yml 
---
foo: !vault |
          $ANSIBLE_VAULT;1.1;AES256
          31316130633062343736316132653837386161353765663836656635626333303133356432363262
          3463663261316238336136313030653439353138636236310a346366313637653063326237626239
          36646461323330363137363530383234393737383065343364623333363337333132633464373463
          3564613266623630360a656166353333343934653136343234636264356366623334383736653639
          3564
~~~

2. Call the above string.yml file inside the playbook by using `include_vars` module

~~~
cat example.yml 
---
- hosts: all 
  tasks:
  - include_vars: string.yml

  - debug:
      msg: "{{ foo }}"
~~~

3. Create a Credential with type vault and store the vault password and then create a Job Template in Tower that calls the above example.yml playbook.

## Expected Result:

Job executes successfully without showing the encrypted value

## Actual Result:

Job executes successfully but shows the encrypted value

~~~
TASK [include_vars] ************************************************************
task path: /var/lib/awx/projects/_39__ansibletower/example.yml:5
Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x7fb5cb96b790> for vault_id=default

Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x7fb5cb96b790> for vault_id=default
 [WARNING]: Failure using method (v2_runner_on_ok) in callback plugin
(<ansible.plugins.callback.awx_display.CallbackModule object at
0x7fb5cd9afed0>): u'bar' is not JSON serializable

Callback Exception: 
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 333, in send_callback
    method(*new_args, **kwargs)
   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/ansible_runner/display_callback/module.py", line 362, in v2_runner_on_ok
    with self.capture_event_data('runner_on_ok', **event_data):
   File "/usr/lib64/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/ansible_runner/display_callback/module.py", line 94, in capture_event_data
    event_context.dump_begin(sys.stdout)
   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/ansible_runner/display_callback/events.py", line 171, in dump_begin
    self.cache.set(":1:ev-{}".format(begin_dict['uuid']), begin_dict)
   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/ansible_runner/display_callback/events.py", line 51, in set
    partial_data = json.dumps(value)
   File "/usr/lib64/python2.7/json/__init__.py", line 243, in dumps
    return _default_encoder.encode(obj)
   File "/usr/lib64/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
   File "/usr/lib64/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
   File "/usr/lib64/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
~~~

**Comment by AlanCoding** [comment]
This seems extremely likely to be impacted by https://github.com/ansible/ansible-runner/pull/331, could you test with the development version of ansible-runner and see where that leaves us?

**Comment by gamuniz** [comment]
@AlanCoding can we pip install it directly into the awx venv?

**Comment by Ompragash** [comment]
@AlanCoding I tested with the development version of ansible-runner and it is not showing the encrypted string and working as expected.

Below are the steps that I followed on Tower server:

~~~

source /var/lib/awx/venv/awx/bin/activate
pip uninstall ansible-runner
git clone git://github.com/ansible/ansible-runner
cd ansible-runner
umask 0022
pip install .
deactivate
~~~

**Comment by matburt** [comment]
We'll mark this one as resolved in Runner 1.4 which is the upcoming release
